src/network.js

```diff
@@ -578,18 +596,17 @@ Network.getWorkerSharedFunctions = function() {
   //  using the .toString() method
 
   // Load and name the train function
-  var train_f = Trainer.prototype.train.toString();
-  train_f = train_f.replace('function (set', 'function train(set') + '\n';
+  var train_f = 'train: ' + Trainer.prototype.train.toString() + ',\n';
```

Comment:
Yes, because the `this` references inside stringfyed methods wont work on the original approach. (Also allowing to build literal objects can help to isolate features and help future uses for "static" or "exported" network code)